todo_app/data/db_service.py

```diff
@@ -21,112 +20,121 @@ def connectDatabase(connectionString):
-            id=id,
-            idShort=idShort,
-        )
-        cardList.append(card)
+        return list_of_cards
 
-    return cardList
+    def get_item(self, taskId: str) -> Card:
+        """
+        Fetches the MongoDB document with the specified ID.
 
+        Args:
+            id: The ID of the item.
+            collection: the MongoDB document collection object
 
-def getItem(collection: pymongo.cursor.Cursor, taskId: str) -> Card:
-    """
-    Fetches the MongoDB document with the specified ID.
+        Returns:
+            card: the card representation of the MongoDB document,
+            or None if no items match the specified ID.
+        """
 
-    Args:
-        id: The ID of the item.
-        collection: the MongoDB document collection object
+        returnedDocument = self.collection.find_one({"_id": ObjectId(taskId)})
 
-    Returns:
-        card: the card representation of the MongoDB document,
-        or None if no items match the specified ID.
-    """
+        try:
 
-    returnedDocument = collection.find_one({"_id": ObjectId(taskId)})
+            id = returnedDocument["_id"]
+            description = returnedDocument["description"]
+            name = returnedDocument["title"]
+            listName = returnedDocument["status"]
+            idShort = str(id)[:7]
 
-    print(f"returned document:{returnedDocument}")
+            card = Card(
+                name=name,
+                description=description,
+                listName=listName,
+                id=id,
+                idShort=idShort,
+            )
 
-    try:
+            return card
 
-        id = returnedDocument["_id"]
-        description = returnedDocument["description"]
-        name = returnedDocument["title"]
-        listName = returnedDocument["status"]
-        idShort = str(id)[:7]
+        # send exception to web server console
+        except Exception as e:
+            print(e)
+            return None
 
-        card = Card(
-            name=name,
-            description=description,
-            listName=listName,
-            id=id,
-            idShort=idShort,
+    def update_task(self, id, status: str) -> pymongo.results.UpdateResult:
+
+        update = self.collection.update_one(
+            {"_id": ObjectId(id)}, {"$set": {"status": f"{status}"}}
         )
 
-        return card
+        return update
+
+    def add_item(self, document) -> pymongo.results.InsertOneResult:
 
-    except Exception as e:
-        print(e)
-        return None
+        result = self.collection.insert_one(document)
+        return result
 
+    def get_user_role(self, userid):
 
-def updateTask(
-    collection: pymongo.cursor.Cursor, id, status: str
-) -> pymongo.results.UpdateResult:
+        user = self.collection.find_one({"userid": userid})
+        if user == None:
+            new_user: pymongo.results.InsertOneResult = self.collection.insert_one(
+                {"userid": userid, "role": "reader"}
```

Comment:
Nice work getting on with this stretch goal! In practice, how do you expect any writer's to be set, will you do it manually via the DB, or should we set it with an environment variable or similar?